heritability/rhemc_pipeline.py

```diff
@@ -0,0 +1,1334 @@
+    in `_import_manifest`.
+
+    ancestries: `list`
+
+    n_include: `int` or None
+    This is primarily used for testing. If a non-None value is provided,
+    this function will trim the dictionary to include n_include items from each
+    trait type.
+
+    random_phenos: `int` or None
+    If not none, then will queue up a bunch of random phenotype jobs. Each of these will
+    be run on all phenotypes.
+
+    suffix: `str`
+    Appended to the end of the results file to search for.
+
+    specific_pheno: `list` or None
+    A list of formatted phenotypes to subset the manifest to. If any are not found, an error will be thrown.
+    """
+    if random_phenos is not None:
+        dct_out = {}
+        for i in range(1, random_phenos+1):
+            pheno_id = RANDOMPREF(i)
+            pheno_file = get_pheno_filename(pheno_id)
+            saige_file = convert_pheno_id_to_potential_saige(pheno_id)
+            dct_out.update({pheno_id: (pheno_file, saige_file, ancestries)})
+        return dct_out
+    else:
+        phenotype_id = manifest.phenotype_id.collect()
+        pheno_file = manifest.pheno_file.collect()
+        saige_file = manifest.saige_file.collect()
+        anc_vec = [[anc for anc in anclist if anc in ancestries] 
+                for anclist in manifest.pop_split.collect()]
+
+        pheno_to_len = {phen: len(anclist) for phen, _, _, anclist in zip(phenotype_id, pheno_file, saige_file, anc_vec)}
+        zipped_vals = zip(phenotype_id, pheno_file, saige_file, anc_vec)
+        dct_out = {id: (file, saigefile, anclist) for id, file, saigefile, anclist in zipped_vals if len(anclist) > 0}
+
+        if n_include is not None:
+            trait_types = list(manifest.aggregate(hl.agg.collect_as_set(manifest.trait_type)))
+            list_of_pheno_lists = []
+            for trait_type in trait_types:
+                corresponding_phenos = manifest.filter(manifest.trait_type == trait_type).phenotype_id.collect()
+                corr_phenos_slim = [pheno for pheno in corresponding_phenos if pheno_to_len[pheno] > 0]
+                list_of_pheno_lists.append(corr_phenos_slim[0:min(n_include, len(corr_phenos_slim))])
+            traits_to_keep = [item for pheno_list in list_of_pheno_lists for item in pheno_list]
+            return {key: dct_out[key] for key in traits_to_keep}
+        elif specific_pheno is not None:
+            if not all([x in dct_out.keys() for x in specific_pheno]):
+                missing_pheno = ', '.join([x for x in specific_pheno if x not in dct_out.keys()])
+                raise ValueError('ERROR: the following phenotypes were not found: ' + missing_pheno)
+            return {key: dct_out[key] for key in specific_pheno}
+        else:
+            return dct_out
+
+
+def _read_pheno_rowtable():
+    """ Reads the row Hail Table from the FULL phenotype MatrixTable.
+    """
+    return _read_pheno_data(False,load_full=True).rows()
+
+
+def _read_pheno_data(checkpoint, load_full=False):
```

Comment:
I'd probably make `checkpoint` default to False here and most places you make this kind of construction

Reply:
Happy to do this, but why? I would think that checkpointing should speed operations up downstream?

Reply:
Well, it's up to you. I like to err on the side of defaulting to never overwriting anything. But if this is just for intermediates that's probably fine

Reply:
I've now gone through all the updated files and made `checkpoint` default to `False` when not writing to intermediate (temporary) file folders.